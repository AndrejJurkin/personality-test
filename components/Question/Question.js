import React from "react";
import Answer from "../Answer/Answer";

export default function Question({
  question,
  selectedAnswerId,
  onAnswerClick,
}) {
  return (
    <>
      <div className="mb-8 font-display">{question.question}</div>
      <div className="flex flex-col gap-2">
        {question.answers.map((answer) => (
          <Answer
            key={answer.id}
            answer={answer}
            onAnswerClick={onAnswerClick}
            selected={answer.id === selectedAnswerId}
          />
        ))}
      </div>
    </>
  );
}
