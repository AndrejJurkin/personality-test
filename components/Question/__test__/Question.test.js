import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Question from "../Question";

describe("Question", () => {
  it("renders a heading", () => {
    const { getByText } = render(<Question question={question} />);
    expect(getByText(/Question Title/i)).toBeInTheDocument();
  });

  it("renders all questions", () => {
    const { getByText } = render(<Question question={question} />);
    question.answers.forEach((answer) => {
      expect(getByText(answer.text)).toBeInTheDocument();
    });
  });

  it("calls onAnswerClick", () => {
    const onAnswerClick = jest.fn();
    const { getByText } = render(
      <Question question={question} onAnswerClick={onAnswerClick} />
    );
    getByText(question.answers[0].text).click();
    expect(onAnswerClick).toHaveBeenCalled();
  });

  it("highlights selected answer", () => {
    const { getByTestId } = render(
      <Question question={question} selectedAnswerId={question.answers[0].id} />
    );
    const selectedAnswer = getByTestId(question.answers[0].id);
    expect(selectedAnswer).toHaveClass("bg-gray-100");
  });

  it("renders component unchanged", () => {
    const { container } = render(
      <Question question={question} selectedAnswerId={question.answers[0].id} />
    );
    expect(container).toMatchSnapshot();
  });
});

const question = {
  id: 1,
  question: "Question Title",
  answers: [
    {
      id: 1,
      text: "Question Answer 1",
      weight: -1,
    },
    {
      id: 2,
      text: "Question Answer 2",
      weight: -1,
    },
    {
      id: 3,
      text: "Question Answer 3",
      weight: 1,
    },
    {
      id: 4,
      text: "Question Answer 4",
      weight: 1,
    },
  ],
};
