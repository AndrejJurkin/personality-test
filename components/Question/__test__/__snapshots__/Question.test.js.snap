// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question renders component unchanged 1`] = `
<div>
  <div
    class="mb-8 font-display"
  >
    Question Title
  </div>
  <div
    class="flex flex-col gap-2"
  >
    <div
      class="cursor-pointer rounded-md border p-3 bg-gray-100"
      data-testid="1"
    >
      <span
        class="select-none text-sm"
      >
        Question Answer 1
      </span>
    </div>
    <div
      class="cursor-pointer rounded-md border p-3"
      data-testid="2"
    >
      <span
        class="select-none text-sm"
      >
        Question Answer 2
      </span>
    </div>
    <div
      class="cursor-pointer rounded-md border p-3"
      data-testid="3"
    >
      <span
        class="select-none text-sm"
      >
        Question Answer 3
      </span>
    </div>
    <div
      class="cursor-pointer rounded-md border p-3"
      data-testid="4"
    >
      <span
        class="select-none text-sm"
      >
        Question Answer 4
      </span>
    </div>
  </div>
</div>
`;
